MerlinAutoUpdate.sh

```diff
@@ -128,26 +134,40 @@ Update_Custom_Settings() {
     esac
 }
 
+##----------------------------------------##
+## Modified by Martinski W. [2023-Oct-05] ##
+##----------------------------------------##
 get_current_firmware() {
-    local current_version="$(nvram get buildno)_$(nvram get extendno)"
+    local current_version="$(nvram get buildno).$(nvram get extendno)"
     echo "$current_version"
 }
 
+##----------------------------------------##
+## Modified by Martinski W. [2023-Oct-05] ##
+##----------------------------------------##
 get_latest_firmware() {
     local url="$1"
-    local page=$(curl -s "$url")
```

Comment:
Direct processing without storing in a variable can be more efficient, especially if the data is used only once. It reduces memory usage (although for a small script like this the impact is negligible). However, if debugging is required, having the data in a separate variable can be helpful. For the purpose of this script, the modification seems logical for now, debugging this step should no longer be required.

Reply:
For debugging purposes, I'd suggest creating a separate "Debug Mode" logfile either on the USB-attached disk drive (if available) or in a temporary folder like "/tmp/var/tmp/" so that you can later review it if/when things go sideways. Having the option to generate a more verbose "Debug Mode" logfile can also be extremely useful when the script is "out in the field" and some user is having problems with the initial setup or the actual F/W update process. In such cases, you can instruct them to turn ON "Debug Mode" and supply the resulting debug log. This can save hours trying to diagnose a problem occurring on a user's specific router environment.

My 2 cents.

Reply:
> For debugging purposes, I'd suggest creating a separate "Debug Mode" logfile either on the USB-attached disk drive (if available) or in a temporary folder like "/tmp/var/tmp/" so that you can later review it if/when things go sideways. Having the option to generate a more verbose "Debug Mode" logfile can also be extremely useful when the script is "out in the field" and some user is having problems with the initial setup or the actual F/W update process. In such cases, you can instruct them to turn ON "Debug Mode" and supply the resulting debug log. This can save hours trying to diagnose a problem occurring on a user's specific router environment.
> 
> My 2 cents.

I really like this idea, it's been noted. 
Thank you again for the help btw, while I have scripting experience this is my first time for a merlin router so the guidance on different values to check for, and seeing examples goes a long way!